app/actions/remote/channel.ts

```diff
@@ -459,3 +460,57 @@ export const switchToChannelByName = async (serverUrl: string, channelName: stri
         return {error};
     }
 };
+
+export async function getChannelMemberCountsByGroup(serverUrl: string, channelId: string, includeTimezones: boolean) {
+    try {
+        const client = NetworkManager.getClient(serverUrl);
+        const channelMemberCountsByGroup = await client.getChannelMemberCountsByGroup(channelId, includeTimezones);
+        return {channelMemberCountsByGroup};
+    } catch (error) {
+        return {error};
+    }
+}
+
+export async function getChannelTimezones(serverUrl: string, channelId: string) {
+    try {
+        const client = NetworkManager.getClient(serverUrl);
+        const channelTimezones = await client.getChannelTimezones(channelId);
+        return {channelTimezones};
+    } catch (error) {
+        return {error};
+    }
+}
+
+export async function makeDirectChannel(serverUrl: string, otherUserId: string, switchToChannelFoo = true) {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    const currentUserId = await queryCurrentUserId(operator.database);
+    const channelName = getDirectChannelName(currentUserId, otherUserId);
+
+    const channel = await queryChannelByName(operator.database, channelName);
+    let result;
+    if (channel) {
+        result = {data: channel};
+    } else {
+        try {
+            const client = NetworkManager.getClient(serverUrl);
+            const newChannel = await client.createDirectChannel([currentUserId, otherUserId]);
+
+            result = {data: newChannel};
+            operator.handleChannel({channels: [newChannel], prepareRecordsOnly: false});
+
+            // Missing membership?
```

Comment:
Not sure whether we should get also the membership and store them, or it is done somehow in a different place.
Also, on V1 there was a check on whether there was a membership for this channel before returning it. It is possible to have the channel in the database but not be a member of the channel? In that case, would it be needed to create the direct channel on the server?

Reply:
so the channel and mychannel are always expected for a DM, if you don't have it and need to created it you would need not only the channel but the channel info as well. Basically I would expect this to populate:
* Channel
* ChannelInfo
* MyChannel
* MyChannelSettings

Reply:
I solved this, but there is potential of improvement in the API to send the membership object along the new channel. Should I create a ticket about that? If so, to which team should it be assigned?

Reply:
About the ticket, yes we should. About the team idk.

In the meantime we could fetch the member

Reply:
https://mattermost.atlassian.net/browse/MM-40426